osbenchmark/workload/params.py

```diff
@@ -964,32 +981,116 @@ def params(self):
             "neighbors": true_neighbors,
         })
         self._update_request_params()
-
-        self.query_params.update({
-            self.PARAMS_NAME_BODY: self._build_vector_search_query_body(self.field_name, vector)})
+        self._update_body_params(vector)
         self.current += 1
         self.percent_completed = self.current / self.total
         return self.query_params
 
-    def _build_vector_search_query_body(self, field_name: str, vector) -> dict:
+    def _build_vector_search_query_body(self, vector) -> dict:
         """Builds a k-NN request that can be used to execute an approximate nearest
         neighbor search against a k-NN plugin index
         Args:
-            field_name: name of field to search
             vector: vector used for query
         Returns:
             A dictionary containing the body used for search query
         """
         return {
-            "size": self.k,
-            "query": {
                 "knn": {
-                    field_name: {
+                    self.field_name: {
                         "vector": vector,
                         "k": self.k
                     }
                 }
             }
+
+
+class BulkVectorsDataSetParamSource(ParamSource):
```

Comment:
Why have this class and `BulkVectorsFromDataSetParamSource`?

Reply:
We want to extend from ParamSource to inherit properties from ParamSource. I used Delegation pattern similar to Bulk param source.

Reply:
But doesnt `BulkVectorsFromDataSetParamSource` extend `VectorDataSetPartitionParamSource` which extends `ParamSource`? Not sure I understand what delegating here helps with.

Reply:
No. VectorDataSetPartitionParamSource is an abstract class which doesn't have any parent

Reply:
I see, is there any reason `VectorDataSetPartitionParamSource` couldnt just extend ParamSource and be abstract? I see I did it this way awhile ago in https://github.com/opensearch-project/k-NN/blame/main/benchmarks/osb/extensions/param_sources.py but Im not sure I remember if there was a good reason.

Reply:
Okay, i updated the class to extend ParamSource. I only have to override property "infinite" instead of setting in constructor.